hil/cli.py

```diff
@@ -151,94 +97,54 @@ def setup_http_client():
-    """
-    if response.status_code < 200 or response.status_code >= 300:
-        sys.stderr.write('Unexpected status code: %d\n' % response.status_code)
-        sys.stderr.write('Response text:\n')
-        sys.stderr.write(response.content + "\n")
-        raise FailedAPICallException()
-    else:
-        sys.stdout.write(response.content + "\n")
-
-# Function object_url should be DELETED.
-
-
-def object_url(*args):
-    """Return a url with a prefix of the HIL endpoint, and args as the
-    (remaining) segments of the path.
-
-    TODO: This function's name is no longer very accurate.  As soon as it is
-    safe, we should change it to something more generic.
-    """
-    # Prefer an environmental variable for getting the endpoint if available.
-    url = os.environ.get('HIL_ENDPOINT')
-    if url is None:
-        config.setup()
-        url = cfg.get('client', 'endpoint')
+###############################
+### First level subcommands ###
+###############################
 
-    for arg in args:
-        url += '/' + urllib.quote(arg, '')
-    return url
+@click.group()
+def cli():
+    """The HIL Command line"""
 
+@cli.group()
+def node():
+    """Commands related to node"""
 
-# Helper functions for making HTTP requests against the API.
-#    Uses the global variable `http_client` to make the request.
-#
-#    Arguments:
-#
-#        `url` - The url to make the request to
-#        `data` - the body of the request (for PUT, POST and DELETE)
-#        `params` - query parameters (for GET)
+@cli.group()
+def project():
+    """Commands related to project"""
 
-def do_put(url, data={}):
-    """do a put request and check the response."""
-    check_status_code(http_client.request('PUT', url, data=json.dumps(data)))
+@cli.group()
+def project():
+    """Commands related to project"""
 
+@cli.group()
+def network():
+    """Commands related to network"""
 
-def do_post(url, data={}):
-    """do a post request and check the response."""
-    check_status_code(http_client.request('POST', url, data=json.dumps(data)))
+@cli.group()
+def switch():
+    """Commands related to switch"""
 
+@cli.group()
+def port():
+    """Commands related to port"""
 
-def do_get(url, params=None):
-    """do a get request and check the response."""
-    check_status_code(http_client.request('GET', url, params=params))
+@cli.group()
+def user():
+    """Commands related to user"""
 
+######################################
+## HIL ADMIN Commands (to be moved) ##
+######################################
 
-def do_delete(url):
-    """do a delete request and check the response."""
-    check_status_code(http_client.request('DELETE', url))
-
-# DELETE UPTIL HERE once all calls have client library support.
-
-
-@cmd
-def version():
-    """Check hil version"""
-    sys.stdout.write("HIL version: %s\n" % VERSION)
-
-
-@cmd
+@cli.command(name='serve', help='Run a development HIL server')
+@click.argument('port', type=click.Choice(str(i) for i in range(1,2**16)),
```

Comment:
Since here we are enforcing a range of port numbers, does it make sense that we are allowing the users to use port numbers under `1000`?

Reply:
I wouldn't worry about this too much because this is meant for development, so I would let developers choose whatever they want.